packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -642,11 +502,94 @@ export class Relayer {
+      this.logger.error({ at: "AcrossRelayer#Disputer", message: "Something errored disputing relay!", error });
     }
   }
 
-  private generateSlowRelayTx(deposit: Deposit, realizedLpFeePct: BN): TransactionType {
+  private async _processTransactionBatch(
+    transactions: { transaction: TransactionType | any; message: string; mrkdwn: string }[]
+  ) {
+    // Remove any undefined transaction objects or objects that contain null transactions.
+    transactions = transactions.filter((transaction) => {
+      if (!transaction) return false;
+      if (!transaction.transaction) return false;
+      return true;
+    });
+
+    if (transactions.length == 0) return;
+    if (transactions.length == 1) {
+      this.logger.debug({ at: "AcrossRelayerTxProcessor", message: "Sending transaction" });
+      await this._sendTransaction(transactions[0].transaction, transactions[0].message, transactions[0].mrkdwn);
+    }
+    if (transactions.length > 1) {
+      // The `to` field in the transaction must be the same for all transactions or the batch processing will not work.
+      // This should be a MultiCaller enabled contract.
+      const targetMultiCallerContract = new this.l1Client.l1Web3.eth.Contract(
+        getAbi("MultiCaller"),
+        transactions[0].transaction._parent._address
+      );
+
+      for (const transaction of transactions) {
+        if (targetMultiCallerContract.options.address != transaction.transaction._parent._address)
+          throw new Error("Batch transaction processing error! Can't specify multiple `to` fields within batch");
+      }
+
+      // Iterate over all transactions and build up a set of multicall blocks and a block of markdown to send to slack
+      // to make the set of transactions readable.
+      const multiCallTransaction = transactions.map((transaction) => transaction.transaction.encodeABI());
+
+      let mrkdwnBlock = "*Transactions sent in batch:*\n";
+      transactions.forEach((transaction) => {
+        mrkdwnBlock += `  • ${transaction.message}:\n`;
+        mrkdwnBlock += `    - ${transaction.mrkdwn}\n`;
+      });
+
+      // Send the batch transaction to the L1 bridge pool contract. Catch if the transaction succeeds.
+      const batchTxSuccess = await this._sendTransaction(
+        (targetMultiCallerContract.methods.multicall(multiCallTransaction) as unknown) as TransactionType,
+        "Multicall Transaction batch sent!🧙",
+        mrkdwnBlock
+      );
+
+      // In the event the batch transaction was unsuccessful, iterate over all transactions and send them individually.
+      if (!batchTxSuccess) {
+        for (const transaction of transactions) {
+          this.logger.info({ at: "AcrossRelayerTxProcessor", message: "Sending batched transactions individually😷" });
+          await this._sendTransaction(transaction.transaction, transaction.message, transaction.mrkdwn);
+        }
+      }
```

Comment:
If the goal is to have bad transactions not block others, should we fire these off using `Promise.allSettled` and then send out errors using `logger.error()` for any that fail? Also, doing these in parallel will probably speed the bot up a lot.

Reply:
in future, yes! that would be great. for now though this won't work as we dont have a robust enough way to deal with nonce collisions for parallel transactions being sent. I spent a few days earlier this week working on a technique to send this kind of transaction. I build out a system were you can "fire and forget" transactions; i.e only wait until they are in the mempool and you have a tx hash and not block for block inclusion. however, I realised that how we detect pending nonces (transactions in the mempool) is not robust enough for this kind of thing. it would _sometimes_ find the pending transactions and other times would not. This is a symptom of how quickly I was sending transactions.

To really implement this kind of logic you need some object that will: a) track the starting nonce b) send transactions and increment it's local nonce c) manage nonce incrementing for the set of transactions.

I'd like to build this out as a full utility that the transaction management offers in the `common` package. 

for now, though, this will just be a bit slow :)